spec/Section 3 -- Type System.md

```diff
@@ -1541,6 +1595,21 @@ is constructed with the following rules:
   definition does not provide a default value, the input object field
   definition's default value should be used.
 
+Further, if the input object is a Oneof Input Object, the following additional
+rules apply:
+
+* If the input object literal or unordered map does not contain exactly one
+  entry, an error must be thrown.
+
+* If the single entry in the input object literal or unordered map is {null},
+  an error must be thrown.
```

Comment:
These are already enforced by validation, so I do not know if we need them in coercion. However, if we do not enforce them here then the coercion table below would have different outputs (e.g. `{a: $a, b: $b}` with `{ "a": "abc" }` would be allowed by coercion, but not by validation).

An alternative approach is to keep the above and add a different non-null coerced value rule so we can drop the two rules below (since they'll be implicitly true).

@leebyron what are your thoughts on this?

Reply:
> These are already enforced by validation, so I do not know if we need them in coercion. However, if we do not enforce them here then the coercion table below would have different outputs (e.g. `{a: $a, b: $b}` with `{ "a": "abc" }` would be allowed by coercion, but not by validation).
> 
> An alternative approach is to keep the above and add a different non-null coerced value rule so we can drop the two rules below (since they'll be implicitly true).
> 
> @leebyron what are your thoughts on this?

@benjie @erickessler1

It looks like [the implementing PR for oneof](https://github.com/graphql/graphql-js/pull/3513) tests allows  `{a: $a, b: $b}` with `{ "a": "abc" }` within coercion, only disallowing in validation. The spec changes as understood here by me seem to disallow this within coercion. I am not sure what the desired behavior was decided to be/should be, but just double checking that everything is aligned. :)

Reply:
see https://github.com/graphql/graphql-js/pull/4034

Reply:
Bumping this